app/components/donation/credit-card.js

```diff
@@ -23,9 +23,7 @@ export default Component.extend({
   cardInvalid: not('cardValid'),
   cardValid: and('isCardNumberValid', 'isCVCValid', 'isExpiryValid'),
 
-  preventSubmit: or('isSubmitting', 'cardInvalid'),
-
-  isSubmitting: false,
+  isBusyOrInvalid: or('isBusy', 'cardInvalid'),
```

Comment:
Route simply has an `isLoading` flag now, when anything is happening, which is passed down into the components. Much simpler that way, with the exception being the card form, which can also be disabled if the card is invalid. Still, simpler than the previous approach.